coalib/misc/Caching.py

```diff
@@ -0,0 +1,122 @@
+import calendar
+import time
+import hashlib
+
+from coalib.misc.CachingUtilities import (
+    pickle_load, pickle_dump, time_consistent, update_time_db,
+    delete_cache_files)
+
+
+class FileCache:
+
+    def __init__(self, log_printer, project_dir, flush_cache=False):
+        """
+        Initialize cache instance for project. Example usage:
+
+        >>> from pyprint.NullPrinter import NullPrinter
+        >>> from coalib.output.printers.LogPrinter import LogPrinter
+        >>> import copy, time
+        >>> log_printer = LogPrinter(NullPrinter())
+
+        To initialize the cache create an instance for the project:
```

Comment:
ah you always need a newline between text and example for it to be rendered correctly